### PR TITLE
use isinstance instead of type comparison

### PR DESCRIPTION
**Pull request by kashif**
# What does this PR do?

Fix for E721 errors which are annoying when doing `make style` with a newer `ruff`

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -156,7 +156,7 @@
     "rhoknp>=1.1.0,<1.3.1",
     "rjieba",
     "rouge-score!=0.0.7,!=0.0.8,!=0.1,!=0.1.1",
-    "ruff>=0.0.241,<=0.0.259",
+    "ruff>=0.0.241",
```
Comment (amyeroberts) [inline setup.py:159]: This shouldn't be changed

**Review by amyeroberts — COMMENTED** [review]
Thanks for all these changes to improve the code! 

Mostly just small nits. Only main comment is the modification of the pinned ruff versions.

File: utils/notification_service.py [hunk 2]
```diff
@@ -20,7 +20,7 @@
 import os
 import re
 import sys
-import time
+from time import sleep
```
[the remark this answers is not in the document]
Comment (kashif) [reply utils/notification_service.py:23]: right the issue was that there is a function `time` defined in the `Message` class which gives a "Redefinition" warning...

**Comment by kashif** [comment]
@amyeroberts thanks! so I relaxed the `ruff` version as with these changes the latest `ruff` runs without raising any issues... but yes i un-did that as per your request

**Review by amyeroberts — APPROVED** [review]
Thanks for these improvements to the code!

**Comment by kashif** [comment]
fixed by #27144

**Pull request closed without merging**